src/components/application_manager/test/commands/mobile/set_app_icon_test.cc

```diff
@@ -144,6 +160,515 @@ TEST_F(SetAppIconRequestTest, OnEvent_UI_UNSUPPORTED_RESOURCE) {
+  EXPECT_CALL(app_mngr_, protocol_handler())
+      .WillRepeatedly(ReturnRef(mock_app_protocol_handler));
+
+  MockProtocolHandlerSettings mock_protocol_handler_settings;
+  EXPECT_CALL(mock_app_protocol_handler, get_settings())
+      .WillRepeatedly(ReturnRef(mock_protocol_handler_settings));
+
+  const protocol_handler::MajorProtocolVersion mock_app_mngr_protocol_version =
+      protocol_handler::PROTOCOL_VERSION_4;
+  EXPECT_CALL(mock_protocol_handler_settings, max_supported_protocol_version())
+      .WillRepeatedly(Return(mock_app_mngr_protocol_version));
+
+  const std::string icon_folder = app_storage_folder + "/" + app_folder;
+  EXPECT_CALL(mock_app_mngr_settings, app_icons_folder())
+      .WillRepeatedly(ReturnRef(icon_folder));
+
+  const uint32_t storage_max_size = 10u;
+  EXPECT_CALL(mock_app_mngr_settings, app_icons_folder_max_size())
+      .WillRepeatedly(ReturnRef(storage_max_size));
+
+  const std::string app_policy_id = "test_icon";
+  EXPECT_CALL((*mock_app), policy_app_id())
+      .WillRepeatedly(Return(app_policy_id));
+
+  const uint32_t app_id = 152u;
+  EXPECT_CALL((*mock_app), app_id()).WillRepeatedly(Return(app_id));
+
+  MessageSharedPtr result;
+  EXPECT_CALL(app_mngr_, ManageHMICommand(_))
+      .WillOnce(DoAll(SaveArg<0>(&result), Return(true)));
+
+  const std::string directory = app_storage_folder + "/" + app_folder;
+  const std::string full_file_path = directory + "/" + kSyncFileName;
+  file_system::CreateDirectoryRecursively(directory);
+  file_system::CreateFile(full_file_path);
+  req->Run();
+  file_system::RemoveDirectory(app_storage_folder);
+
+  const std::string expect_sync_file_name =
+      file_system::ConvertPathForURL(full_file_path);
+  const std::string actual_sync_file_name =
+      (*result)[am::strings::msg_params][am::strings::sync_file_name]
+               [am::strings::value].asString();
+  EXPECT_EQ(expect_sync_file_name, actual_sync_file_name);
+
+  const uint32_t expect_app_id = app_id;
+  const uint32_t actual_app_id = static_cast<uint32_t>(
+      (*result)[am::strings::msg_params][am::strings::app_id].asInt());
+  EXPECT_EQ(expect_app_id, actual_app_id);
+}
+
+TEST_F(SetAppIconRequestTest, Run_NonExistentApp_ApplicationNotRegistered) {
+  utils::SharedPtr<SetAppIconRequest> req =
+      CreateCommand<SetAppIconRequest>(msg_vr);
+
+  MockAppPtr mock_app;
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+
+  MessageSharedPtr result;
+  EXPECT_CALL(app_mngr_, ManageMobileCommand(_, _))
+      .WillOnce(DoAll(SaveArg<0>(&result), Return(true)));
+
+  req->Run();
+
+  const mobile_apis::Result::eType expect_result_code =
+      mobile_apis::Result::APPLICATION_NOT_REGISTERED;
+  const mobile_apis::Result::eType actual_result_code =
+      static_cast<const mobile_apis::Result::eType>(
+          (*result)[am::strings::msg_params][am::strings::result_code].asInt());
+  EXPECT_EQ(expect_result_code, actual_result_code);
```

Comment:
@BSolonenko in this case such check is the overhead. You could compare incoming result code just inside EXPECT_CALL like:
```
EXPECT_CALL(app_mngr_, ManageMobileCommand(mobile_apis::Result::APPLICATION_NOT_REGISTERED, _)) 
```

Reply:
@AKalinich-Luxoft The first parameter of the method `ManageMobileCommand` takes `SmartObjectSPtr`.